### PR TITLE
[FormRecognizer] Final documentation updates before 4.0.0

### LINKED ISSUE #24488: Terminology for Prebuilt Document Model
We are currently using "Prebuilt Document Model" but should it say "general"? 
- Prebuilt General Document Model
- General Prebuilt Document Model
- General Document Model

### LINKED ISSUE #26782: [FormRecognizer] Add section or comment about build modes to the migration guide
Currently, the v2 to v3 migration guide does not include any mentions to the addition of the build mode required parameter when building custom models. This is a major breaking change in usage, so it should be addressed in the guide.

### LINKED ISSUE #30476: [FormRecognizer] Add locale documentation in docstrings
Since `Locale` is now a string, users need to follow documentation to understand what locales are supported.

### PR DESCRIPTION
**Pull request by kinelski**
Changes included in this PR:
- **Public API change:** moved `DocumentAnalysisClient`-related extension methods defined in `FormRecognizerClientBuilderExtensions` to a separate `DocumentAnalysisClientBuilderExtensions` class. This was a last-minute change requested by architects before approval, but impact is minimal.
- Updated documentation to refer to the SDK version as `4.0.0` instead of `4.0.0-beta.x`.
- Updated documentation to refer to the service version as `2022-08-31`.
- Added a section about changes in long-running operation service methods design to the migration guide.
- Added a paragraph about build modes to the migration guide.
- Removed any remaining mention to "entities" since this feature is not supported anymore.
- Updated "Get and List operations" sample description to explain about the hierarchy in `OperationDetails` and its subtypes.
- Update terminology used for the "prebuilt general document" model to align with the term used by the service.
- Other minor changes.

Fixes #30476
Fixes #26782
Fixes #24488

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Azure;
+using Azure.AI.FormRecognizer.DocumentAnalysis;
+using Azure.Core.Extensions;
+
+namespace Microsoft.Extensions.Azure
+{
+    /// <summary>
+    /// Extension methods to add the <see cref="DocumentAnalysisClient"/> to clients builder.
+    /// </summary>
+    public static class DocumentAnalysisClientBuilderExtensions
```
Comment (pallavit) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientBuilderExtensions.cs:14]: Where are these extensions used?

**Review by pallavit — APPROVED** [review]
:shipit:

File: sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 4.0.0-beta.6 (Unreleased)
+## 4.0.0 (2022-09-08)
```
Comment (pallavit) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md:3]: Ship it!

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientBuilderExtensions.cs (hunk 1 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientBuilderExtensions.cs:14]: They're used for integration with ASP.NET Core. That's described in our [guidelines](https://azure.github.io/azure-sdk/dotnet_implementation.html#integration-with-aspnet-core).

**Pull request merged**